### PR TITLE
Fix nations never building cities on very crowded maps 🏙️

### PR DESCRIPTION
**Pull request by FloPinguin**
## Description:

On games with high nation density (many nations relative to land tiles, e.g. 400 nation games), nations would build Ports/Factories forever and never build a single City.

NationStructureBehavior has a special case that lets a nation build a Port (or Factory if landlocked) as its very first structure instead of a City, so it can get income sooner on crowded maps. This was gated on unitsOwned(City) === 0. Since that branch never builds a City itself, that condition never became false, so it kept re-triggering on every structure decision the nation ever made, permanently blocking it from reaching the actual City-building code further down.

The fix gates this branch on its own one-shot flag, consumed only once the branch actually succeeds in placing a structure. A shared counter like placementsCount doesn't work here: nations that are both high-starting-gold Hard/Impossible and on a crowded map already have an unrelated branch that builds a SAM launcher as their literal first structure, which would consume a shared "first structure" slot and cause this branch to be skipped entirely for those nations. The dedicated flag keeps retrying on later calls if an attempt fails (e.g. temporarily unaffordable) and fires independently of whatever else the nation has already built, while still only ever succeeding once.

Noticed this problem by watching https://www.youtube.com/watch?v=709JPXBXfno

Screenshot 2026-09-10 224519
Screenshot 2026-09-10 225041

## Please complete the following:

- [X] I have added screenshots for all UI updates
- [X] I process any text displayed to the user through translateText() and I've added it to the en.json file
- [X] I have added relevant tests to the test directory

## Please put your Discord username so you can be contacted if a bug or regression is found:

FloPinguin

### CODE REVIEW

File: tests/NationStructureBehavior.test.ts [hunk 1]
```diff
@@ -855,6 +855,88 @@ describe("NationStructureBehavior.countDefensePostsNearFront", () => {
...
+        isUnitDisabled: () => false,
+        gameConfig: () => ({
+          difficulty: opts.difficulty ?? Difficulty.Medium,
+        }),
+        startingGold: () => 0n,
+      }),
+      sharedWaterComponents: () => null, // landlocked -> Factory preferred
+      nations: () => Array(400).fill({}),
+      numLandTiles: () => 1_000_000, // 400 / 1_000_000 > 1/7500 -> high density
+    };
```
[the remark this answers is not in the document]
Comment (FloPinguin) [reply tests/NationStructureBehavior.test.ts:880]: I checked this against the current code before acting on it — the suggestion isn't something I'm going to make, and I want to say why rather than just doing it.

Verified, and declining, with reasons:

The nations() list is real map data, not a mockable count. GameImpl.nations() returns _nations, populated from a map's manifest nation-spawn list (see [GameImpl.ts:367](vscode-webview://1cvl9ebu1946gnntca3pkhgo9tu19cm66q6tk8692v7hs8pa3t9s/src/core/execution/nation/NationStructureBehavior.ts)). To hit the density threshold this bug needs (nations/landTiles > 1/7500), a real setup() test would need a fixture map with ~400 defined nation spawn points — none of the existing test maps have anywhere near that, so this isn't a small swap, it's a new large fixture.

The file's own convention is unit-level mocking of private decision methods. All 59 pre-existing tests in this file (shouldUseConnectivityScore, buildReachableStations, tryBuildDefensePost, sampleTilesNearFront, getAttackFrontTiles, countDefensePostsNearFront) test NationStructureBehavior's private methods directly against synthetic game/player objects — none use setup(). My 3 added tests match that established pattern, not a new one.

The bug is in a boolean-guard condition, not in placement mechanics. Whether ConstructionExecution actually places a Factory/City/SAM correctly is already covered by other test files (ConstructionExecution, ConstructionCost, etc.). What needed testing here is specifically: does the decision to call maybeSpawnStructure fire once, not forever, and not get starved by the SAM branch — which is exactly what mocking maybeSpawnStructure isolates. Driving it through real handleStructures() would also drag in defense-post gating, cooldown/save-up-phase logic, and real gold/tile economy — all irrelevant noise for this regression, and a source of flakiness.

**Review by evanpelle — APPROVED** [review]
